### ISSUE TITLE
No BaseStepListener POST XX not registered, while iterating values in parallel stream and to result.

### ISSUE THREAD

**Issue by CuriousQA** [serenity-bdd/serenity-core#2870]
The output of the following code(using Streams) is reported only for the main thread but not for the Fork commonPools. Is there any way that this gets reported in Serenity reports or should we develop a custom report based on that?

`        result.parallelStream().map(x ->  {

            Response respToken = SerenityRest.given()
                    .pathParam("username", username)
                    .pathParam("password", password)
                    .when()
                    .post(authentication);
            String resToken = respToken .jsonPath().getJsonObject("token"); 
             Serenity.reportThat("call resToken : " + resToken ,
                    () -> Assertions.assertThat(resToken ).isEqualToIgnoringCase("ABCD")
            );
           return null;
    }).collect(Collectors.toList());`

Output:

`17:25:52.038 [ForkJoinPool.commonPool-worker-4] INFO net.serenitybdd.rest.decorators.request.RequestSpecificationDecorated - No BaseStepListener, POST request1 not registered.
17:25:52.038 [ForkJoinPool.commonPool-worker-3] INFO net.serenitybdd.rest.decorators.request.RequestSpecificationDecorated - No BaseStepListener, POST request2 not registered.
17:25:52.038 [ForkJoinPool.commonPool-worker-2] INFO net.serenitybdd.rest.decorators.request.RequestSpecificationDecorated - No BaseStepListener, POST request3 not registered.`

Basically the request1, request2 & request3 are same but with change in value that we get from Streams.
The output is shown only for main thread execution but not for the ForkJoinPool.commonPool-worker-{1,2,3,4,5,6}

**Comment by wakaleo** [comment]
Possibly fixed in 3.2.1
